cookiecutter-django modifications to align with our django-kedro project

millsks opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Updating cookiecutter with prompts and package management that align to our project.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here to auto close when merged -->
<!-- e.g., closes #xxx or resolves #xxx -->
Resolves ResponsibleAIML/django-kedro#12

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Identify features and improvements in the cookiecutter-django package that aligns with our project.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
The new features and any modifications will be included in the unit tests.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

millsks committed:
Update pyproject.toml with poetry dependencies

millsks committed:
Move redis to use_celery check

millsks committed:
Update kedro logic in pyproject.toml cookiecutter.json

millsks committed:
Test kedro package presence in baked cookie.

millsks committed:
Remove print statements from test_kedro_in_pyproject

millsks committed:
Remove print statements from test_kedro_in_pyproject

millsks committed:
Merge remote-tracking branch 'origin/feature/cookiecutter-poetry' into feature/cookiecutter-poetry

elanqo commented on tests/test_cookiecutter_generation.py:
```diff
@@ -362,6 +364,19 @@ def test_pycharm_docs_removed(cookies, context, use_pycharm, pycharm_docs_exist)
+    kedro_package_check_list = ["kedro", "kedro-datasets", "kedro-airflow", "kedro-mlflow"]
+
+    if use_kedro == "y":
+        assert all([x in pyproject_contents for x in kedro_package_check_list]) == True
```
Instead of if else, you can have the result as part of parametrize.

millsks replied:
I was using if/else because of the any and all functions being used depending on the value of use_kedro.  I am open to alternatives though!  What did you have in mind?

millsks replied:
I will also look into the lock file as part of the cookiecutter, but it might need to be added by the end user in their repo because it would be generated post cookie bake?

elanqo replied:
Something like:
`@pytest.mark.parametrize("use_kedro, result", [("n", False), ("y", True)])`

Then assert
`assert all([x in pyproject_contents for x in kedro_package_check_list]) == result`

elanqo reviewed (commented):
Looks good for me. I do recommend commiting  .lock file, for build consistency.

millsks commented:
> Looks good for me. I do recommend commiting  .lock file, for build consistency.

I will also look into the lock file as part of the cookiecutter, but it might need to be added by the end user in their repo because it would be generated post cookie bake?

millsks committed:
Result.project property is deprecated

millsks committed:
Update cookiecutter to use the poetry pdm

millsks committed:
Add message for poetry install

millsks commented:
poetry can handle the [extra dependenc](https://python-poetry.org/docs/dependency-specification/#extras).

> Extras and @ can be combined as one might expect (package[extra]@version):
> `poetry add django[bcrypt]@^4.0.0`

millsks committed:
Add kedro-viz to dependency list

millsks committed:
Add pre-commit and show commands for poetry

millsks committed:
chdir to cookiecutter.project_slug prior to pre-commit hook install

millsks committed:
Initializing git repo in baked project

millsks committed:
Validate and autoupdate pre-commit hooks

millsks committed:
Move precommit commands to own function and control with use_precommit response

millsks committed:
Update managers.py to use typing.Union

Annotations are not supported until Python 3.10.  Updating managers.py to use typing.Union instead of "|" allows us to use cookiecutter-django with Python 3.9.

millsks committed:
Keep env files whether using docker/heroku or local environment.

millsks committed:
Keep env files only if using docker/heroku.

millsks committed:
Modify site migration script to ignore the site id sequence update when sqlite is being used

millsks committed:
Add option to run 'npm install' if Webpack was the frontend option

millsks committed:
Use sqlite3 db for django by default if DATABASE_URL is not set

millsks committed:
Fix typo in default database definition in base.py

millsks committed:
Update database connection details in production.py

millsks committed:
Simplified cookiecutter template by removing package management calls

millsks committed:
Remove kedro, poestry, and precommit from the cookiecutter-django-kedro-serving.json replay file

millsks committed:
Update pytest-xdist to >=3.3.1

millsks committed:
Package nodejs with cookiecutter

millsks committed:
Package nodejs with cookiecutter after removing .git subfolder.

millsks committed:
Package nodejs with cookiecutter after removing .git subfolder.

millsks committed:
Copy the node directory only without rendering any variables

millsks committed:
Update sample replay file

Remove the _output_dir keyand added the _copy_without_render key.

millsks committed:
Add flake8 plugins to project pyproject.toml

millsks committed:
Add node to .gitignore

millsks committed:
Add node/alias and node/versions to .gitignore

millsks committed:
Update node related files to ignore
